app/services/users/omniauth.rb

```diff
@@ -1,36 +1,24 @@
-class Users::Omniauth
-  attr_reader :auth
-
-  def initialize(auth)
-    @auth = auth
-  end
-
+class Users::Omniauth < Struct.new(:auth)
   def get
+    find_or_create_authorization
     user
   end
 
   private
 
-  delegate :provider, :uid, :info, to: :auth
-  delegate :email, to: :info
-
-  def user
-    @_user ||= find_user || create_user
+  def find_or_create_authorization
+    user.authorizations
+        .create_with(uid: uid)
+        .find_or_create_by(provider: provider)
```

Comment:
@sinpai let's use two spaces indentation
```ruby
user.authorizations
  .create_with(uid: uid)
  .find_or_create_by(provider: provider)
```

If `user` will be changed to `person` or something similar -> we'll need to change 3 lines with your solution